client/components/allProducts.js

```diff
@@ -0,0 +1,56 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import {ProductListing} from './productListing'
+
+class allProducts extends Component {
+  //CHECK ASSUMPTION: FETCHPRODUCTS IN MAIN'S COMPONENT DID MOUNT?
+  constructor(props) {
+    super(props)
+    this.state = {
+      isClicked: false,
+      isAdmin: false
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  //POTENTIAL VIEW PRODUCT BUTTON
+  handleClick() {
+    this.setState(state => ({isClicked: !state.isClicked}))
```

Comment:
perhaps we also want to push to the history here (using withRouter from react-router-dom), so that once the product is created, we navigate to the new product's page? maybe not, though, if the assumption is that an admin might want to add more than one product.